libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -30,50 +33,76 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
     const auto vote_hash = vote->getHash();
     LOG(log_dg_) << "Received PBFT vote " << vote_hash;
 
+    // Check this before we are marking vote as seen to have an opportunity to process it after
+    if (vote->getRound() > pbft_mgr_->getPbftRound() + kVoteAcceptingRounds) {
```

Comment:
this will also not work in our new design with periods & rounds

Reply:
I guess we should properly implement a class to store (period, round) and then we can implement comparison operators > , < , and such    But in this case logic would be I think if period is equal then this check is ok, else if period is one higher, then vote->getRound() > kVoteAcceptingRounds, or if period is higher by more than one.